docs/development/plugin.md

```diff
@@ -0,0 +1,188 @@
+# Plugins
+
+Plugins is a way to extend launchr functionality.  
+The main difference from actions is that plugins must be written in `go` and run natively on a host machine.
+
+## Available plugins
+
+Available plugin types:
+
+1. `OnAppInitPlugin` - runs on the application init stage.
+2. `ActionDiscoveryPlugin` - discovers and provides actions
+3. `CobraPlugin` - adds cobra cli functionality. Use `ActionDiscoveryPlugin` to add new actions.
+4. `PersistentPreRunPlugin` - runs before cobra cobra command when all arguments are parsed.
+5. `GeneratePlugin` = generates supporting files before build.
+
+Plugin implementation examples:
+
+1. [Default plugins](../plugins)
+2. [Keyring](https://github.com/launchrctl/keyring)
+3. [Compose](https://github.com/launchrctl/compose)
+4. [Web](https://github.com/launchrctl/web)
+
```

Comment:
We could also list these 2:
- https://github.com/launchrctl/plugin-example-plugin-runtime
- https://github.com/launchrctl/plugin-example-container-runtime